src/DynamoCoreWpf/Controls/DynamoTextBox.cs

```diff
@@ -390,26 +390,20 @@ public void SetDataContext(object dataContext, bool closeImmediately = false)
             // If Dynamo window is not active, we should not show as well as hide tooltip or do any other staff.
             if (mainDynamoWindow == null || !mainDynamoWindow.IsActive) return;
 
-            if (dataContext == null)
+            if (closeImmediately)
             {
-                if (closeImmediately)
-                {
-                    CloseLibraryToolTipPopup(null, null);
-                    return;
-                }
-                showTimer.Stop();
-                dispatcherTimer.Start();
+                CloseLibraryToolTipPopup();
                 return;
             }
-            dispatcherTimer.Stop();
-            nextDataContext = dataContext;
-            showTimer.Start();
+
+            toolTipTimer.Start(dataContext, 150);
         }
 
-        private void OpenLibraryToolTipPopup(object sender, EventArgs e)
+        private void OpenCloseLibraryToolTipPopup(object dataContext)
         {
-            this.DataContext = nextDataContext;
-            IsOpen = true;
+            IsOpen = dataContext != null || this.IsMouseOver;
+            if (dataContext != null)
+                this.DataContext = dataContext;
```

Comment:
I'm always curious about this, what if we always set `this.DataContext` to `dataContext` (even if it is `null`), does that make any difference?